### PR TITLE
Split ES types into separate indices (for ES 6+)

### LINKED ISSUE #292: Split ElasticSearch documents into separate indices
https://github.com/elastic/elasticsearch/issues/15613

Starting from v6.0, ElasticSearch will stop supporting multiple types in an index.

### PR DESCRIPTION
**Pull request by mh-park**
elastic/elasticsearch#15613

Future versions (v6.0) of ElasticSearch will prevent new indices from being created with more than one type.

Resolves #292

### CODE REVIEW

File: plugin/storage/es/spanstore/reader_test.go [hunk 1]
```diff
@@ -629,7 +629,7 @@ func mockSearchService(r *spanReaderTest) *mock.Call {
 	searchService.On("Aggregation", stringMatcher(servicesAggregation), mock.AnythingOfType("*elastic.TermsAggregation")).Return(searchService)
 	searchService.On("Aggregation", stringMatcher(operationsAggregation), mock.AnythingOfType("*elastic.TermsAggregation")).Return(searchService)
 	searchService.On("Aggregation", stringMatcher(traceIDAggregation), mock.AnythingOfType("*elastic.TermsAggregation")).Return(searchService)
-	r.client.On("Search", mock.AnythingOfType("string"), mock.AnythingOfType("string"), mock.AnythingOfType("string"), mock.AnythingOfType("string")).Return(searchService)
+	r.client.On("Search", mock.AnythingOfType("string"), mock.AnythingOfType("string"), mock.AnythingOfType("string"), mock.AnythingOfType("string"), mock.AnythingOfType("string")).Return(searchService)
```
Comment (mh-park) [inline plugin/storage/es/spanstore/reader_test.go]: This will fix #290

Comment (mh-park) [inline plugin/storage/es/spanstore/reader_test.go]: @black-adder

File: plugin/storage/es/spanstore/reader.go [hunk 2]
```diff
@@ -160,33 +161,33 @@ func (s *SpanReader) unmarshalJSONSpan(esSpanRaw *elastic.SearchHit) (*jModel.Sp
 }
 
 // Returns the array of indices that we need to query, based on query params
-func findIndices(startTime time.Time, endTime time.Time) []string {
+func findIndices(typeOfIndex string, startTime time.Time, endTime time.Time) []string {
```
Comment (black-adder) [inline plugin/storage/es/spanstore/reader.go]: s/typeOfIndex/prefix?

**Comment by yurishkuro** [comment]
nice!

**Pull request merged**